### PR TITLE
Add new tutorial

### PR DESCRIPTION
**Pull request by gdangelo**
Magic Link Authentication in Next.js with NextAuth and Fauna

## Changes 💡

I added a new tutorial for passwordless authentication with NextAuth + Fauna DB, with live example + code source.

## Affected issues 🎟

None.

## Screenshot (If Applicable) 📷

None.

### CODE REVIEW

File: versioned_docs/version-v3/tutorials.md [hunk 1]
```diff
@@ -51,6 +51,23 @@ How to use `useSession()` hook with class components.
...
+
+Learn how to implement passwordless/magic link authentication with database storage in your Next.js projects using NextAuth and Fauna DB.
+
+> The final version of the project's code can be found on [Github](https://github.com/AlterClassIO/magic-next-auth). You can use it as a starting point for any Next.js app that requires passwordless authentication.
+
+> You can also preview the example live [here](https://magic-next-auth.vercel.app/).
+
+This tutorial covers:
+
+- Configuring Next.js, NextAuth, and Fauna to work together seamlessly
```
Comment (balazsorban44) [inline versioned_docs/version-v3/tutorials.md:64]:
```suggestion
- Configuring Next.js, NextAuth.js, and Fauna to work together seamlessly
```

File: versioned_docs/version-v3/tutorials.md [hunk 2]
```diff
@@ -51,6 +51,23 @@ How to use `useSession()` hook with class components.
...
+- Using Next.js dynamic API routes to handle authentication requests
+- Using Fauna and the Fauna Adapter for next-auth to persist users, email sign in tokens, and sessions
```
Comment (balazsorban44) [inline versioned_docs/version-v3/tutorials.md:66]:
```suggestion
- Using Fauna and the Fauna Adapter for `next-auth` to persist users, email sign in tokens, and sessions
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Hi, NextAuth.js v4 has recently been launched with quite a few changes, could you update your blog post?

https://next-auth.js.org/getting-started/upgrade-v4

**Comment by gdangelo** [comment]
I just updated the tutorial with NextAuth.js v4 🔥

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Could you move this to https://github.com/nextauthjs/docs/blob/0fbb96a8835fab29f06034b13ee9e1966981795a/docs/tutorials.md

**Comment by gdangelo** [comment]
Done! Sorry, I didn't realize I was still on the v3 docs ;)

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**